src/main/java/com/codejoust/main/model/problem/ProblemContainer.java

```diff
@@ -0,0 +1,42 @@
+package com.codejoust.main.model.problem;
+
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity // This tells Hibernate to make a table out of this class
+@Getter
+@Setter
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+public class ProblemContainer {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Integer id;
+    
+    // The problem associated with this game report
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "problem_table_id")
+    private Problem problem;
+
+    // The number of users who solved this problem.
+    private Integer userSolvedCount;
+
+    // The number of test cases.
+    private Integer testCaseCount;
+
+    // The average (mean) number of test cases passed.
+    private Double averageTestCasesPassed;
+
+    // The average (mean) number of submission attempts.
+    private Double averageAttemptCount;
```

Comment:
These values are set to be stored in the database; @alankbi, thoughts on that? We will want to store some things like `testCaseCount` because the number of `testCases` on the problem could change in the future.

The reason I stored the other statistics too was because I figured it was better to store the information than re-calculate it every time, as this is a relatively large number of calculations. However, I'm open to other arguments, especially if it is unorthodox to store statistic information like this in a database - let me know your thoughts!

Reply:
I think that sounds good - based on our convo in standup a few days ago, I think that makes sense to avoid having to constantly recalcuate those